editor/src/frontend/frontend_message_handler.rs

```diff
@@ -11,6 +12,7 @@ pub enum FrontendMessage {
 	CollapseFolder { path: Vec<LayerId> },
 	ExpandFolder { path: Vec<LayerId>, children: Vec<LayerPanelEntry> },
 	SetActiveTool { tool_name: String },
+	SetToolOptions { tool_name: String, tool_options: Option<ToolOptions> },
```

Comment:
Both of these are fair points, but I think they're each probably worthy of a separate discussion/PR.

I agree that each tool should have a ToolOption, in which case a `HashMap` may not be the ideal interface. At a minimum, we could probably have a getter function that uses `entry(...).or_insert(...)` so it looks like the map is always complete.

Merging `FrontendMessage::SetToolOptions` with `FrontendMessage::SetActiveTool` would have a minor side-effect. Namely, it would make it impossible to communicate a tool option change to the frontend without selecting that tool. For now, it would be fine to make the tool options frontend "lazy" like that, but I could see that functionality being useful in the future.